.docker/alpine/lean/Dockerfile

```diff
@@ -0,0 +1,70 @@
+# This is an alternative Dockerfile for `leanprover/lean`,
+# based on `alpine`, and installs `elan` and `leanproject`.
+# See also the image `leanprover/mathlib` which contains a copy of mathlib.
+
+# `elan` doesn't include a release for `x86_64-unknown-linux-musl`,
+# so we roll our own
+FROM clux/muslrust as elan-builder
+
+RUN git clone https://github.com/Kha/elan.git .
+RUN cargo build
+
+FROM alpine as elan-install
+
+RUN adduser -D lean
+USER lean
+WORKDIR /home/lean
+COPY --from=elan-builder /volume/target/x86_64-unknown-linux-musl/debug/elan-init .
+RUN ./elan-init -y && rm elan-init
+
+# We start with a temporary `builder` container, from https://github.com/six8/pyinstaller-alpine
+# in which we install `python` and prepare a standalone version of `leanproject`.
+FROM six8/pyinstaller-alpine:alpine-3.6-pyinstaller-v3.4 as leanproject-builder
+
+USER root
+# install prerequisites
+RUN apk update && apk add --no-cache python3 py3-pip py3-virtualenv
+# create a non-root user
+RUN adduser -D lean
+
+USER lean
+WORKDIR /home/lean
+
+ENV PATH="/home/lean/.local/bin:$PATH"
+
+# install `leanproject` using `pip`
+RUN python3 -m pip install --user mathlibtools
+
+# run `pyinstaller` on the installed copy of `leanproject`
+WORKDIR /home/lean/.local/bin
+RUN /pyinstaller/pyinstaller.sh --onefile --noconfirm leanproject
+# this has created `/home/lean/.local/bin/dist/leanproject`,
+# which we'll now copy to a fresh container
+
+# Now we start the `alpine` build.
+FROM alpine
+
+USER root
+# install prerequisites
+# (leanproject assumes bash)
```

Comment:
Does it? Where?

Reply:
The wonders of reproducible builds! Removing `bash` from `apk add ...` results in 
```
Step 3/3 : RUN leanproject get mathlib
 ---> Running in 8c4901ce58ee
Cloning from git@github.com:leanprover-community/mathlib.git
Error cloning via SSH, trying HTTPS...
Cloning from https://github.com/leanprover-community/mathlib.git
info: downloading component 'lean'
info: installing component 'lean'
env: can't execute 'bash': No such file or directory
Command '['leanpkg', 'configure']' returned non-zero exit status 127.
```
when building the `leanprover/mathlib:alpine` container.

Reply:
It sounds like this is actually `leanpkg` that requires `bash`? That looks like there's a shell script somewhere with `#!env bash` or something in its title, which `leanproject` doesn't contain.

Reply:
Looks like it is `https://github.com/leanprover/lean/blob/master/bin/leanpkg`. In any case, remove `bash` as a dependency saves <2mb, so I don't think there's much point trying to avoid it. I'll change the comment.

Reply:
```suggestion
# (leanpkg assumes bash)
```